tools/kagami/src/main.rs

```diff
@@ -211,6 +234,63 @@ mod genesis {
         result.transactions[0].isi.push(mint.into());
         Ok(result)
     }
+
+    fn generate_synthetic(
+        domains: u64,
+        accounts_per_domain: u64,
+        assets_per_domain: u64,
+    ) -> color_eyre::Result<RawGenesisBlock> {
+        let mut builder = RawGenesisBlockBuilder::new();
+        for domain in 0..domains {
+            let mut domain_builder = builder.domain(format!("domain_{domain}").parse()?);
+
+            for account in 0..accounts_per_domain {
+                domain_builder = domain_builder.with_account(
+                    format!("account_{account}").parse()?,
+                    DEFAULT_PUBLIC_KEY.parse()?,
+                );
+            }
+
+            for asset in 0..assets_per_domain {
+                domain_builder = domain_builder
+                    .with_asset(format!("asset_{asset}").parse()?, AssetValueType::Quantity);
+            }
+
+            builder = domain_builder.finish_domain();
+        }
+        let mut genesis = builder.build();
+
+        let mints = {
```

Comment:
These aren't strictly necessary, but since we have them, we should consider adding `Transfer`s in a separate issue.